File: blind-nav-web/src/types/alert.ts
export type AlertType =
  | 'CRITICAL_REQUEST_CREATED'
  | 'SLA_AT_RISK'
  | 'SLA_BREACHED'
  | 'ESCALATION_TRIGGERED'
  | 'REQUEST_UNASSIGNED';

export type AlertSeverity = 'INFO' | 'WARNING' | 'HIGH' | 'CRITICAL';

export interface Alert {
  id: string;
  type: AlertType;
  severity: AlertSeverity;
  title: string;
  message: string;
  requestId?: string;
  createdAt: string;
  isRead: boolean;
  source: string;
}

export interface SlaStats {
  activeRequests: number;
  onTrack: number;
  atRisk: number;
  breached: number;
  escalated: number;
  criticalOpen: number;
  unreadAlerts: number;
}

File: blind-nav-backend/app/schemas/alert.py
from pydantic import BaseModel
from typing import Optional


class Alert(BaseModel):
    id: str
    type: str        # CRITICAL_REQUEST_CREATED | SLA_AT_RISK | SLA_BREACHED | ESCALATION_TRIGGERED | REQUEST_UNASSIGNED
    severity: str    # INFO | WARNING | HIGH | CRITICAL
    title: str
    message: str
    requestId: Optional[str] = None
    createdAt: str
    isRead: bool = False
    source: str = "system"   # 'system' | 'rules' | 'llm'
